### PR TITLE
Load ITelemetryProcessors when setting up the TelemetryConfigration

### LINKED ISSUE #3741: ITelemetryProcessor does not appear to be supported in Function Apps
_From @StevenTaylor on October 5, 2018 5:50_

Documentation (https://docs.microsoft.com/en-us/azure/application-insights/app-insights-api-filtering-sampling) suggests that ITelemetryProcessor can be used to filter\sample the telemetry before it is sent App Insights.

But the implementation code is never called in our Function App.

The code to set up the builder is called, which looks like:

`                var builder = TelemetryConfiguration.Active.TelemetryProcessorChainBuilder;            
                builder.Use((next) => new AppInsightsFilter(next));                                
                builder.Build();`

But the 'Process' method implemented in the class is never called, the class looks like:

`    class AppInsightsFilter : ITelemetryProcessor
    {
        private ITelemetryProcessor Next { get; set; }

        // Link processors to each other in a chain.
        public AppInsightsFilter(ITelemetryProcessor next)
        {
            this.Next = next;
        }

        public void Process(ITelemetry item)
        {
            // To filter out an item, just return
            if (!OKtoSend(item)) { return; }

            // Modify the item if required
            //ModifyItem(item);

            this.Next.Process(item);
        }

        private bool OKtoSend(ITelemetry item)
        {            
            // try something!
            return (DateTime.Now.Second <= 20); // only send for the first 20 seconds of a minute
        }
    }`

PS. We have tried setting config values SamplingPercentage (App Insights config) and maxTelemetryItemsPerSecond (host.json) as low as possible to reduce the telemetry data, but there is still too much telemetry.

_Copied from original issue: Azure/Azure-Functions#981_

### PR DESCRIPTION
**Pull request by luthus**
Should fix https://github.com/Azure/azure-functions-host/issues/3741

### CODE REVIEW

**Review by Steg17 — APPROVED** [review]

**Comment by Greybird** [comment]
Hello,

Any chance to have this fix merged ?

**Comment by brettsam** [comment]
@luthus -- I know this is over a year old. Are you still available to push this through? If not, I can add a simple unit test here and get it in for you.

**Comment by brettsam** [comment]
Pinging @cijothomas here as well for another set of eyes.

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -352,6 +354,14 @@ internal static LoggerFilterOptions CreateFilterOptions(LoggerFilterOptions regi
                 configuration.TelemetryProcessorChainBuilder.UseSnapshotCollector(options.SnapshotConfiguration);
             }
 
+            if (telemetryProcessorFactories.Any())
+            {
+                foreach (ITelemetryProcessorFactory telemetryProcessorFactory in telemetryProcessorFactories)
+                {
+                    configuration.TelemetryProcessorChainBuilder.Use(telemetryProcessorFactory.Create);
```
Comment (cijothomas) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs:361]: https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/NETCORE/src/Shared/Implementation/TelemetryConfigurationOptionsSetup.cs#L104

ApplicationInsights adds processors to default sink. Not sure if its applicable here.

**Comment by liliankasem** [comment]
@brettsam @luthus @AnatoliB is this something we still want to get merged? Looks like just one more comment to address

**Comment by RohitRanjanMS** [comment]
This approach won’t work as the ITelemeteryProcessorFactory is in Microsoft.ApplicationInsights.AspNetCore and this is marked as private in runtimeAssemblies. Will have to think of another approach to fix this.

**Pull request closed without merging**